plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePartitionManager.java

```diff
@@ -109,27 +109,39 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
                 .map(HiveColumnHandle::getType)
                 .collect(toList());
 
+        Optional<List<String>> partitionNames = hiveTableHandle.getPartitionNames();
         Iterable<HivePartition> partitionsIterable;
         Predicate<Map<ColumnHandle, NullableValue>> predicate = constraint.predicate().orElse(value -> true);
+        boolean partitionsAreLoaded = hiveTableHandle.getPartitions().isPresent();
         if (hiveTableHandle.getPartitions().isPresent()) {
             partitionsIterable = hiveTableHandle.getPartitions().get().stream()
                     .filter(partition -> partitionMatches(partitionColumns, effectivePredicate, predicate, partition))
                     .collect(toImmutableList());
         }
         else {
-            List<String> partitionNames = getFilteredPartitionNames(metastore, tableName, partitionColumns, compactEffectivePredicate);
-            partitionsIterable = () -> partitionNames.stream()
+            List<String> partitionNamesList = hiveTableHandle.getPartitionNames()
+                    .orElse(getFilteredPartitionNames(metastore, tableName, partitionColumns, compactEffectivePredicate));
+            partitionsIterable = () -> partitionNamesList.stream()
                     // Apply extra filters which could not be done by getFilteredPartitionNames
                     .map(partitionName -> parseValuesAndFilterPartition(tableName, partitionName, partitionColumns, partitionTypes, effectivePredicate, predicate))
                     .filter(Optional::isPresent)
                     .map(Optional::get)
                     .iterator();
+            partitionsAreLoaded = partitionNamesList.size() <= maxPartitions || constraint.predicate().isPresent();
```

Comment:
Partitions are not loaded, since it's a throw-away information.
It's conceivable that `hiveTableHandle` is later consumed with a slightly different logic, making the `enforcedTupleDomain` untrue.